packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -2,20 +2,921 @@
+  final TextStyle? headerHintStyle;
+
+  /// The color of the divider on the search view.
+  ///
+  /// If this property is null, then [SearchViewThemeData.dividerColor] is used.
+  /// If that is also null, the default value is [ColorScheme.outline].
+  final Color? dividerColor;
+
+  /// Optional size constraints for the search view.
+  ///
+  /// If null, the value of [SearchViewThemeData.constraints] will be used. If
+  /// this is also null, then the constraints defaults to:
+  /// ```dart
+  /// const BoxConstraints(minWidth: 360.0, minHeight: 240.0)
+  /// ```
+  final BoxConstraints? viewConstraints;
+
+  /// Called to create a widget which can open a search view route when it is tapped.
+  ///
+  /// The widget returned by this builder is faded out when it is tapped.
+  /// At the same time a search view route is faded in.
+  ///
+  /// This must not be null.
+  final AnchorBuilder anchorBuilder;
+
+  /// Called to get the suggestion list for the search view.
+  ///
+  /// By default, the list returned by this builder is laid out in a [ListView].
+  /// To get a different layout, use [viewBuilder] to override.
+  final SuggestionsBuilder suggestionsBuilder;
+
+  @override
+  State<SearchAnchor> createState() => _SearchAnchorState();
+}
+
+class _SearchAnchorState extends State<SearchAnchor> {
+  bool _anchorIsVisible = true;
+  final GlobalKey _anchorKey = GlobalKey();
+  bool get _viewIsOpen => !_anchorIsVisible;
+  late SearchController? _internalSearchController;
+  SearchController get _searchController => widget.searchController ?? _internalSearchController!;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.searchController == null) {
+      _internalSearchController = SearchController();
+    }
+    _searchController._attach(this);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _searchController._detach(this);
+    _internalSearchController = null;
+  }
+
+  void _openView() {
+    Navigator.of(context).push(_SearchViewRoute(
+      viewLeading: widget.viewLeading,
+      viewTrailing: widget.viewTrailing,
+      viewHintText: widget.viewHintText,
+      viewBackgroundColor: widget.viewBackgroundColor,
+      viewElevation: widget.viewElevation,
+      viewSurfaceTintColor: widget.viewSurfaceTintColor,
+      viewSide: widget.viewSide,
+      viewShape: widget.viewShape,
+      viewHeaderTextStyle: widget.headerTextStyle,
+      viewHeaderHintStyle: widget.headerHintStyle,
+      dividerColor: widget.dividerColor,
+      viewConstraints: widget.viewConstraints,
+      showFullScreenView: getShowFullScreenView(),
+      toggleVisibility: toggleVisibility,
+      textDirection: Directionality.of(context),
+      viewBuilder: widget.viewBuilder,
+      anchorKey: _anchorKey,
+      searchController: _searchController,
+      suggestionsBuilder: widget.suggestionsBuilder,
+    ));
+  }
+
+  void _closeView(String? selectedText) {
+    if (selectedText != null) {
```

Comment:
Does this mean that if nothing was selected the caller will see a stale value?

Reply:
I think the way you have it now is OK; it is pretty clear what's going to happen when closeView(null) is called.